example_project_app/plugins.py

```diff
@@ -275,3 +275,25 @@ def perform_project_modify(
                     action=action.lower(),
                 ),
             )
+
+    def validate_form_app_settings(self, app_settings, project=None, user=None):
+        """Example implementation for app setting validation plugin API"""
+        errors = {}
+        for setting_name in app_settings:
```

Comment:
Now, since you are correctly providing the values provided by the user in the form here, you should also do some checks on the values in this example implementation. Then test against some unaccepted values in your unit tests.